tests/integration/targets/web_app_pool/tasks/tests.yml

```diff
@@ -0,0 +1,423 @@
+  register: restart_from_stop_pool_actual_check
+
+- name: assert restart from stopped web pool check
+  assert:
+    that:
+    - restart_from_stop_pool_check is changed
+    - restart_from_stop_pool_actual_check.stdout == 'Stopped\r\n'
+
+- name: restart from stopped web pool
+  web_app_pool:
+    name: '{{test_iis_webapppool_name}}'
+    state: restarted
+  register: restart_from_stop_pool
+
+- name: get actual status of pool
+  ansible.windows.win_command: powershell.exe "Import-Module WebAdministration; (Get-Item -Path IIS:\AppPools\{{test_iis_webapppool_name}}).state"
+  register: restart_from_stop_pool_actual
+
+- name: assert restart from stopped web pool
+  assert:
+    that:
+    - restart_from_stop_pool is changed
+    - restart_from_stop_pool_actual.stdout == 'Started\r\n'
+
+- name: set web pool attribute that is a collection (check mode)
+  web_app_pool:
+    name: '{{test_iis_webapppool_name}}'
+    state: present
+    attributes:
+      recycling.periodicRestart.schedule: "00:10:00,10:10:00"
+  register: collection_change_check
+  check_mode: yes
+
+- name: get result of set web pool attribute that is a collection (check mode)
+  ansible.windows.win_shell: |
+    Import-Module WebAdministration
+    (Get-ItemProperty -Path "IIS:\AppPools\{{test_iis_webapppool_name}}" -Name recycling.periodicRestart.schedule).Collection | ForEach-Object { $_.value.ToString() }
+  register: collection_change_result_check
+
+- name: assert results of set web pool attribute that is a collection (check mode)
+  assert:
+    that:
+    - collection_change_check is changed
+    - collection_change_result_check.stdout == ""
+
+- name: set web pool attribute that is a collection
+  web_app_pool:
+    name: '{{test_iis_webapppool_name}}'
+    state: present
+    attributes:
+      recycling.periodicRestart.schedule: "00:10:00,10:10:00"
+  register: collection_change
+
+- name: get result of set web pool attribute that is a collection
+  ansible.windows.win_shell: |
+    Import-Module WebAdministration
+    (Get-ItemProperty -Path "IIS:\AppPools\{{test_iis_webapppool_name}}" -Name recycling.periodicRestart.schedule).Collection | ForEach-Object { $_.value.ToString() }
+  register: collection_change_result
+
+- name: assert results of set web pool attribute that is a collection
+  assert:
+    that:
+    - collection_change is changed
+    - collection_change_result.stdout_lines == [ "00:10:00", "10:10:00" ]
+
+- name: set web pool attribute that is a collection (idempotent)
+  web_app_pool:
+    name: '{{test_iis_webapppool_name}}'
+    state: present
+    attributes:
+      recycling.periodicRestart.schedule: [ "00:10:00", "10:10:00" ]
+  register: collection_change_again
+
+- name: assert results of set web pool attribute that is a collection (idempotent)
+  assert:
+    that:
+    - collection_change_again is not changed
+
+# The following tests are only for IIS versions 8.0 or newer
+- block:
```

Comment:
We can remove this block and the when on line 423 and always run these tests as IIS 8 will be present on Server 2016 and newer.